File: 2452. words-within-two-edits-of-dictionary.java
class Solution {
    boolean hasMatch(String query, String[] dictionary) {
        boolean found = false;
        for (String dict : dictionary) {
            int diffMax = 2;
            for (int i=0;i<dict.length();i++) {
                if (dict.charAt(i) != query.charAt(i)) {
                    diffMax--;
                }
                if (diffMax < 0) {
                    break;
                }
            }
            if (diffMax >= 0) {
                return true;
            }
        }

        return found;
    }

    public List<String> twoEditWords(String[] queries, String[] dictionary) {

        List<String> result = new ArrayList<>();
        for (String query : queries) {
            if (hasMatch(query, dictionary)) {
                result.add(query);
            }
        }

        return result;
    }
    
    public static void main(String[] args) {
        Solution sol = new Solution();
        
        // Example 1
        String[] queries1 = {"word","note","ants","wood"};
        String[] dictionary1 = {"wood","joke","moat"};
        List<String> output1 = sol.twoEditWords(queries1, dictionary1);
        System.out.println("Example 1: queries=[\"word\",\"note\",\"ants\",\"wood\"], dictionary=[\"wood\",\"joke\",\"moat\"]");
        System.out.println("Output: " + output1);
        System.out.println("Expected: [word, note, wood]");
        System.out.println();
        
        // Example 2
        String[] queries2 = {"yes"};
        String[] dictionary2 = {"not"};
        List<String> output2 = sol.twoEditWords(queries2, dictionary2);
        System.out.println("Example 2: queries=[\"yes\"], dictionary=[\"not\"]");
        System.out.println("Output: " + output2);
        System.out.println("Expected: []");
    }
}

File: 2452. words-within-two-edits-of-dictionary.py
class Solution:
    def twoEditWords(self, queries: List[str], dictionary: List[str]) -> List[str]:
        def _get_distance(s1,s2):
            cnt = 0
            for i in range(len(s1)):
                if s1[i]!=s2[i]:
                    cnt+=1
                if cnt==3:
                    return False
            return True

        good = []
        for query in queries:
            for d in dictionary:
                dist = _get_distance(query,d)
                if dist:
                    good.append(query)
                    break
        return good

if __name__ == "__main__":
    sol = Solution()
    
    # Example 1
    queries1 = ["word","note","ants","wood"]
    dictionary1 = ["wood","joke","moat"]
    output1 = sol.twoEditWords(queries1, dictionary1)
    print(f"Example 1: queries={queries1}, dictionary={dictionary1}")
    print(f"Output: {output1}")
    print(f"Expected: ['word','note','wood']")
    print()
    
    # Example 2
    queries2 = ["yes"]
    dictionary2 = ["not"]
    output2 = sol.twoEditWords(queries2, dictionary2)
    print(f"Example 2: queries={queries2}, dictionary={dictionary2}")
    print(f"Output: {output2}")
    print(f"Expected: []")